gtda/homology/rescaling.py

```diff
@@ -0,0 +1,333 @@
+    for :math:`j`). If the entries in `X` represent point clouds, their
+    distance matrices are first computed, and then rescaled according to the
+    same formula.
+
+    Parameters
+    ----------
+    metric : string or callable, optional, default: ``'euclidean'``
+        If set to ``'precomputed'``, each entry in `X` along axis 0 is
+        interpreted to be a distance matrix. Otherwise, entries are
+        interpreted as feature arrays, and `metric` determines a rule with
+        which to calculate distances between pairs of instances (i.e. rows)
+        in these arrays.
+        If `metric` is a string, it must be one of the options allowed by
+        :func:`scipy.spatial.distance.pdist` for its metric parameter, or a
+        metric listed in :obj:`sklearn.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`,
+        including "euclidean", "manhattan" or "cosine".
+        If `metric` is a callable function, it is called on each pair of
+        instances and the resulting value recorded. The callable should take
+        two arrays from the entry in `X` as input, and return a value
+        indicating the distance between them.
+
+    metric_params : dict or None, optional, default: ``None``
+        Additional keyword arguments for the metric function.
+
+    neighbor_rank : int, optional, default: ``1``
+        Rank of the neighbors used to modify the metric structure according
+        to the "consistent rescaling" procedure.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means
+        using all processors.
+
+    Attributes
+    ----------
+    effective_metric_params_ : dict
+        Dictionary containing all information present in `metric_params`.
+        If `metric_params` is ``None``, it is set to the empty dictionary.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from gtda.homology import ConsistentRescaling
+    >>> X = np.array([[[0, 0], [1, 2], [5, 6]]])
+    >>> cr = ConsistentRescaling()
+    >>> X_rescaled = cr.fit_transform(X)
+    >>> print(X_rescaled.shape)
+    (1, 3, 3)
+
+    See also
+    --------
+    VietorisRipsPersistence
+
+    References
+    ----------
+    .. [1] T. Berry and T. Sauer, "Consistent manifold representation for
+           topological data analysis"; *Foundations of data analysis* **1**,
+           pp. 1--38, 2019; doi: `10.3934/fods.2019001
+           <http://dx.doi.org/10.3934/fods.2019001>`_.
+
+    """
+    _hyperparameters = {'neighbor_rank': [int, (1, np.inf)]}
+
+    def __init__(self, metric='euclidean', metric_params=None, neighbor_rank=1,
+                 n_jobs=None):
+        self.metric = metric
+        self.metric_params = metric_params
+        self.neighbor_rank = neighbor_rank
+        self.n_jobs = n_jobs
+
+    def _consistent_homology_distance(self, X):
```

Comment:
In connection with my comment on `ConsecutiveRescaling`, perhaps "homology" should be removed from here too and this can be called "_consistent_rescaling" or "_consistent_distance_rescaling"?